src/lcgs/ir/symbol_checker.rs

```diff
@@ -47,19 +52,19 @@ impl<'a> SymbolChecker<'a> {
         }
     }
 
-    /// Checks and evaluates an expressions. This is typically only used in [CheckMode::Const]
+    /// Checks and evaluates an expressions. This is only used in [CheckMode::Const]
     /// where we assume the expression can be reduced to a value already during symbol checking.
-    pub fn check_eval(&self, expr: &Expr) -> Result<i32, ()> {
+    pub fn check_eval(&self, expr: &Expr) -> Result<i32, SymbolError> {
         let checked = self.check(expr)?;
         if let ExprKind::Number(n) = checked.kind {
             Ok(n)
         } else {
-            Err(())
+            panic!("Constant expression was not reduced to a single number.")
```

Comment:
Shouldn't this be an error?

Reply:
Nope. The previous phase should have caught this error already. If it is not a single number here, something is wrong within our compiler.